### PR TITLE
Delete process Related documents when deleting a process. Added process text deletion back-end. Indicate the type of associated documents

### PR DESCRIPTION
**Pull request by rmsamitha**
Fixing https://wso2.org/jira/browse/PC-76
Fixing Fixing https://wso2.org/jira/browse/PC-84

### CODE REVIEW

File: modules/extensions/process/publisher/process/themes/default/partials/list-assets.hbs [hunk 1]
```diff
@@ -47,7 +47,7 @@
             }
         });
 
-        $.ajax({
+        /*$.ajax({
```
Comment (vinok88) [inline modules/extensions/process/publisher/process/themes/default/partials/list-assets.hbs]: Shall we remove the commented?

Comment (rmsamitha) [inline modules/extensions/process/publisher/process/themes/default/partials/list-assets.hbs]: Fixed with #164

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 2]
```diff
@@ -1625,6 +1625,71 @@ public void deleteFlowchart(String name, String version, String user) throws Pro
         }
     }
 
+    /**
+     * Delete the resource collection (directory) created in the governance registry for the process to keep its documents
+     *
+     * @param processName
```
Comment (pamod) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Better put a description for the params

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 3]
```diff
@@ -1625,6 +1625,71 @@ public void deleteFlowchart(String name, String version, String user) throws Pro
...
+        }
+    }
+
+    /**
+     * Delete process related text
+     *
+     * @param processName
+     * @param processVersion
+     * @return
+     * @throws ProcessCenterException
```
Comment (Niranjan-K) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Need to explain about each params for JavaDocs.

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 2 above)
Comment (rmsamitha) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Thanks @pamod. Fixed with 00a11304a5de537269f0cfd08cd1905045196624

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 3 above)
Comment (rmsamitha) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Thanks @Niranjan-K. Fixed with 00a1130

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 4]
```diff
@@ -1668,6 +1733,45 @@ public String updateDescription(String descriptionDetails, String user) throws P
...
+                }
+                deleteDocumentResourceCollection(processName, processVersion);
+
+                //delete other associations
+                removeBPMNDiagram(processName, processVersion, user);
+                deleteFlowchart(processName, processVersion, user);
+                deleteProcessText(processName, processVersion);
+            }
+        } catch (Exception e) {
+            String errMsg =
```
Comment (mohanvive) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: What is the reason for catching the generic exception ?

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 5]
```diff
@@ -1625,6 +1625,71 @@ public void deleteFlowchart(String name, String version, String user) throws Pro
...
+    /**
+     * Delete the resource collection (directory) created in the governance registry for the process to keep its documents
+     *
+     * @param processName
+     * @param processVersion
+     * @throws ProcessCenterException
+     */
+    public void deleteDocumentResourceCollection(String processName, String processVersion)
+            throws ProcessCenterException, RegistryException {
+
```
Comment (sinthuja) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Remove unnecessary blank lines.

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 6]
```diff
@@ -1625,6 +1625,71 @@ public void deleteFlowchart(String name, String version, String user) throws Pro
...
+     * @return
+     * @throws ProcessCenterException
+     */
+    public String deleteProcessText(String processName, String processVersion) throws ProcessCenterException {
+        String processId = "";
+        try {
+            RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+            if (registryService != null) {
+                UserRegistry reg = registryService.getGovernanceSystemRegistry();
+                String processTextPath = ProcessCenterConstants.PROCESS_TEXT_PATH + processName + "/" + processVersion;
```
Comment (sinthuja) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Path separator can be read from constants.

File: modules/extensions/process/publisher/process/themes/default/partials/view-asset.hbs [hunk 7]
```diff
@@ -519,6 +519,7 @@
             <table class="table table-bordered" id="docTable">
                 <thead>
                 <tr>
+                    <th>Type</th>
```
Comment (sinthuja) [inline modules/extensions/process/publisher/process/themes/default/partials/view-asset.hbs]: Why not using locales in the UI segments??

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java [hunk 8]
```diff
@@ -1625,6 +1625,71 @@ public void deleteFlowchart(String name, String version, String user) throws Pro
...
+                    String newProcessContent = xmlToString(processXML);
+                    processResource.setContent(newProcessContent);
+                    reg.put(processPath, processResource);
+
+                    Resource storedProcessAsset = reg.get(processPath);
+                    processId = storedProcessAsset.getUUID();
+                }
+            }
+        } catch (Exception e) {
+            String errMsg = "Error has been occurred while removing BPMN diagram in the process:" + processName + "-"
```
Comment (rmsamitha) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Thanks @mohanvive. The xmlToString method called in the try block throws 'Exception'. So has to catch 'Exception'. And the issue related to logging and throwing exception same time, is fixed already with 636d316f1c9c92034a9d3c7c1a8fef26dbd5ba1b

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 4 above)
Comment (rmsamitha) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Thanks @mohanvive for pointing out the issue. Fixed with 98b499b6273a21f443e4edbd49b29672d8740e3b

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 5 above)
Comment (rmsamitha) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Thanks @sinthuja. Fixed with ec7bff601a782d66a0af8e73a8b6c764c5626a65

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java (hunk 6 above)
Comment (rmsamitha) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java]: Thanks @sinthuja. Fixed with f6732be26542e91b3ee895b7eee9797e423a46da

**Pull request merged**